app/src/main/kotlin/io/sweers/catchup/ui/about/CollapsingAboutHeader.kt

```diff
@@ -64,353 +42,143 @@ import androidx.compose.ui.text.UrlAnnotation
 import androidx.compose.ui.unit.lerp
-import androidx.compose.ui.unit.sp
 import androidx.core.graphics.drawable.toBitmap
-import dev.zacsweers.catchup.compose.CatchUpTheme
-import dev.zacsweers.catchup.compose.arcLerp
 import io.sweers.catchup.R
 import io.sweers.catchup.base.ui.BackPressNavButton
-import io.sweers.catchup.util.UiUtil
-import kotlin.math.abs
-import kotlin.math.absoluteValue
 import kotlin.math.roundToInt
-
-private const val FADE_PERCENT = 0.75F
-private const val TITLE_TRANSLATION_PERCENT = 0.50F
+import me.onebone.toolbar.CollapsingToolbarScaffold
+import me.onebone.toolbar.ScrollStrategy
+import me.onebone.toolbar.rememberCollapsingToolbarScaffoldState
 
 // TODO
-//  - Figure out why it's not settling at targetY.
-//    - Possibly missing accounting for the parallax effect of the resizing column
 //  - Can we use lerp() in more places?
-//  - Can we use lerp() with a custom FastOutSlowIn interpolator?
-//  - Clean up leftover local state vars
+//  - Use arcLerp() for title's movement.
 //  - Can all this state be hoisted?
 //  - Title initially jumps multiple pixels when collapsing
-@OptIn(
-  ExperimentalTextApi::class,
-  ExperimentalMaterial3Api::class,
-)
+@OptIn(ExperimentalTextApi::class)
 @Composable
 fun CollapsingAboutHeader(
   versionName: String,
-  scrollBehavior: TopAppBarScrollBehavior,
   modifier: Modifier = Modifier,
-  maxHeight: Dp = 275.0.dp,
-  pinnedHeight: Dp = 56.0.dp,
+  body: @Composable () -> Unit = {},
 ) {
-  var debugUiCount by remember { mutableIntStateOf(0) }
-  val debugUi = debugUiCount >= 5
-  val pinnedHeightPx: Float
-  val maxHeightPx: Float
-  val top: Int
-  val topDp: Dp
-  LocalDensity.current.run {
-    top = WindowInsets.systemBars.getTop(this)
-    topDp = top.toDp()
-    pinnedHeightPx = pinnedHeight.toPx() + top
-    maxHeightPx = maxHeight.toPx() + top
-  }
-
-  // Sets the app bar's height offset limit to hide just the bottom title area and keep top title
-  // visible when collapsed.
-  SideEffect {
-    if (scrollBehavior.state.heightOffsetLimit != pinnedHeightPx - maxHeightPx) {
-      scrollBehavior.state.heightOffsetLimit = pinnedHeightPx - maxHeightPx
-    }
-  }
-
-  // Set up support for resizing the top app bar when vertically dragging the bar itself.
-  val appBarDragModifier =
-    if (!scrollBehavior.isPinned) {
-      Modifier.draggable(
-        orientation = Orientation.Vertical,
-        state =
-          rememberDraggableState { delta ->
-            scrollBehavior.state.heightOffset = scrollBehavior.state.heightOffset + delta
-          },
-        onDragStopped = { velocity ->
-          settleAppBar(
-            scrollBehavior.state,
-            velocity,
-            scrollBehavior.flingAnimationSpec,
-            scrollBehavior.snapAnimationSpec
+  val scaffoldState = rememberCollapsingToolbarScaffoldState()
+  val parallaxRatio = 0.2f
+
+  Surface(modifier) { // todo: why is this needed when AboutScreen is already using Scaffold()?
```

Comment:
I didn't get a chance to investigate this further

Reply:
What happens without it?

Reply:
IIRC the header's background color and its content colors were black on black